src/server/cluster/cluster_family.cc

```diff
@@ -401,11 +409,15 @@ void ClusterFamily::ReadWrite(CmdArgList args, ConnectionContext* cntx) {
 }
 
 void ClusterFamily::DflyCluster(CmdArgList args, ConnectionContext* cntx) {
-  if (!IsClusterEnabledOrEmulated()) {
-    return cntx->SendError(kClusterDisabled);
+  if (!(IsClusterEnabled() || (IsClusterEmulated() && cntx->journal_emulated))) {
+    return cntx->SendError("Cluster is disabled. Use --cluster_mode=yes to enable.");
   }
 
-  VLOG(2) << "Got DFLYCLUSTER command (" << cntx->conn()->GetClientId() << "): " << args;
+  if (cntx->conn()) {
+    VLOG(2) << "Got DFLYCLUSTER command (" << cntx->conn()->GetClientId() << "): " << args;
+  } else {
+    VLOG(2) << "Got DFLYCLUSTER command (NO_CLIENT_ID): " << args;
```

Comment:
In which flow do you get null connection?

Reply:
oh this is the journal emulated mode

Reply:
nit: 
client_info = cntx->conn()? ntx->conn()->GetClientId() : "stub";
VLOG(2) << "Got DFLYCLUSTER command (" << client_info << args;

Reply:
in journal_emulated